3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js

```diff
@@ -15,12 +18,14 @@ const rollDice = require('../../helpers/pokerDiceRoller');
 function rollTheDices() {
   // TODO Refactor this function
   const dices = [1, 2, 3, 4, 5];
-  return rollDice(1);
+  const myAllPromises = dices.map((element) => rollDice(element));
+  Promise.all(myAllPromises)
+    .then((results) => {
+      console.log('Resolved!', results);
+    })
+    .catch((error) => console.log('Rejected!', error.message));
 }
-
-rollTheDices()
-  .then((results) => console.log('Resolved!', results))
-  .catch((error) => console.log('Rejected!', error.message));
+rollTheDices();
```

Comment:
Can you make it work using the lines 21-23 that you now deleted? So, return the result from `Promise.all()` (in itself a promise) from `rollTheDices()`.

Reply:
I fixed that part as you said.